ipa-core/src/protocol/dp/insecure.rs

```diff
@@ -1,18 +1,43 @@
 #![allow(dead_code)]
 
-use std::f64;
+use std::f64::consts::E;
 
-use rand::distributions::Distribution;
+use rand::distributions::{BernoulliError, Distribution};
 use rand_core::{CryptoRng, RngCore};
 
-use crate::protocol::dp::distributions::{BoxMuller, RoundedBoxMuller};
+use crate::protocol::dp::distributions::{BoxMuller, RoundedBoxMuller, TruncatedDoubleGeometric};
 
-#[derive(Debug, thiserror::Error)]
+#[derive(Debug, PartialEq, thiserror::Error)]
 pub enum Error {
     #[error("Epsilon value must be greater than {}, got {0}", f64::MIN_POSITIVE)]
     BadEpsilon(f64),
     #[error("Valid values for DP-delta are within {:?}, got: {0}", f64::MIN_POSITIVE..1.0 - f64::MIN_POSITIVE)]
     BadDelta(f64),
+    #[error(
+        "Valid values for TruncatedDoubleGeometric are greater than {:?}, got: {0}",
+        f64::MIN_POSITIVE
+    )]
+    BadS(f64),
+    #[error(
+        "Valid values for success probability in Geometric are greater than {:?}, got: {0}",
+        f64::MIN_POSITIVE
+    )]
+    BadGeometricProb(f64),
+    #[error(
+        "Shift value over 1M -- likely don't need it that large and preventing to avoid any chance of overflow
+        in Double Geometric sample",
+    )]
+    BadShiftValue(u32),
+    #[error(
+        "Sensitivity value over 1M -- likely don't need it that large and preventing to avoid any chance of overflow
+        in Double Geometric sample",
```

Comment:
I didn't see any risk of overflow until i32::MAX.  I would instead say "Sensitivity is higher than supported"  (Putting a 1M in this sort of message is a recipe for getting it wrong.  If you want the value in the message, I would use the constant instead, as above.)